activerecord/lib/active_record/store.rb

```diff
@@ -76,31 +100,135 @@ class << self
     end
 
     module ClassMethods
-      def store(store_attribute, options = {})
-        serialize store_attribute, IndifferentCoder.new(options[:coder])
-        store_accessor(store_attribute, options[:accessors]) if options.has_key? :accessors
+      # Defines store on this model.
+      #
+      # +store_name+ The name of the store.
+      #
+      # ==== Options
+      # The following options are accepted:
+      #
+      # +coder+ The coder of the store.
+      #
+      # +accessors+ An array of the accessors to the store.
+      #
+      # Examples:
+      #
+      #   class User < ActiveRecord::Base
+      #     store :settings, accessors: [ :color, :homepage, login_at: :datetime ], coder: JSON
+      #   end 
+      def store(store_name, options = {})
+        serialize store_name, IndifferentCoder.new(options[:coder])
+        store_accessor(store_name, *options[:accessors]) if options.has_key? :accessors
       end
 
-      def store_accessor(store_attribute, *keys)
+      # Adds additional accessors to an existing store on this model.
+      #
+      # +store_name+ The name of the store.
+      #
+      # +keys+ The array of the accessors to the store.
+      #
+      # +typed_keys+ The key-to-type hash of the accesors with type to the store.
+      #
+      # Examples:
+      #
+      #   class SuperUser < User
+      #     store_accessor :settings, :privileges, login_at: :datetime
+      #   end 
+      def store_accessor(store_name, *keys, **typed_keys)
+        _prepare_local_stored_attributes(store_name)
+
         keys = keys.flatten
+        typed_keys = typed_keys.except(keys)
 
         _store_accessors_module.module_eval do
           keys.each do |key|
             define_method("#{key}=") do |value|
-              write_store_attribute(store_attribute, key, value)
+              write_store_attribute(store_name, key, value)
             end
 
             define_method(key) do
-              read_store_attribute(store_attribute, key)
+              read_store_attribute(store_name, key)
             end
           end
         end
 
+        self.local_stored_attributes[store_name] |= keys
```

Comment:
You mean smth like `_prepare_local_stored_attributes(store_name, keys)`? Reasonable.